analyses/survival-analysis/util/survival_models.R

```diff
@@ -0,0 +1,114 @@
+  install.packages("survminer")
+}
+
+# Attach this package
+library(survminer)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+survival_analysis <- function(metadata, ind_var, test = "kap.meier", ind_data = NULL) {
+  # Given the overall survival information, and an independent variable will
+  # run survival analysis and return a list with 1) the model fit object and
+  # 2) a summary table 3) the data used.
+  #
+  # Args:
+  #   metadata: a data.frame that contains columns OS_status and OS_days to use
+  #             for the survival model. This also assumes "LIVING" and "DECEASED"
+  #             are the two statuses. THis will be converted to a numeric variable
+  #             for use with `survival` R functions. Samples with NAs are dropped.
+  #   ind_var: a character string noting the name of the independent variable to
+  #            test as a predictor for survival.
+  #   test: a character string noting which test to use. Supported choices:
+  #         "kap.meier", "log.rank", "cox.reg"
+  #   ind_data: If the data for the independent variable needed for the test is not
+  #         in metadata, add it here. This assumes it will be a data.frame with
+  #         either a "Tumor_Sample_Barcode", "Kids_First_Biospecimen_ID", or
+  #         "sample" column which to inner_join with metadata by.
+  #
+  # Returns:
+  # A list with three objects: 1) the original model fit object,
+  #                            2) the summary table
+  #                            3) the original data.frame
+
+  # List the tests
+  supported_tests <- c("kap.meier", "log.rank", "cox.reg")
+
+  # Check that it is a supported test
+  if (!(test %in% supported_tests)) {
+    stop(paste(test, "is not a supported test. Please specify one of the following:",
+               paste(supported_tests, collapse=", ")))
+  }
+
+  # Format the OS_status variable for the survival model functions
+  metadata$OS_status <- factor(metadata$OS_status, levels = c("LIVING", "DECEASED"))
+  metadata$OS_status_num <- as.numeric(metadata$OS_status)
+
+  # If other ind_data has been supplied, attempt to join it
+  if (!is.null(ind_data)) {
+    # Make it so the column names are automatically matching no matter what
+    colnames(ind_data) <- gsub("Tumor_Sample_Barcode|sample", "Kids_First_Biospecimen_ID", colnames(ind_data))
+
+    # Join this ind_data to the metadata
+    metadata <- dplyr::inner_join(metadata, ind_data,
+      by = "Kids_First_Biospecimen_ID"
+    )
+  }
+
+  # Extract independent variables
+  ind_var_df <- metadata %>%
+    dplyr::select("Kids_First_Biospecimen_ID", "OS_status_num", "OS_days", ind_var)
+
+  # For the model need a plus sign for separating multiple independent variables
+  ind_var <- paste0(ind_var, collapse = "+")
+
+  # Piece together a model
+  model <- paste("survival::Surv(OS_days, OS_status_num)",
+    ind_var,
+    sep = " ~ "
+  )
+  
+  # We'll need to grab this later for ggsurvplot
+  assign("model", model, envir = .GlobalEnv)
```

Comment:
So, is the expectation that a variable named `model` will be propagated to the global environment and then whoever uses the function should be able to go forth with their plotting? I would think it's safer to return this from the function - is that not possible?
If it's not possible to return the model object, we shouldn't send a variable called `model` into the global environment. Too much potential to interfere with existing objects, since it's such a general name.  I'd change the global name to `survival_analysis_model`? Or, could users pass in as a named argument what they want the output model to be named?

Reply:
Yes. This is my janky work around for trying to still get `ggsurvplot` to work. I'm not happy with the current solution (for the list of problems you've noted) but I was trying to figure out what is a better solution that `ggsurvplot` will still be able to use. I can have it save a `model` as another item to the list that is returned if that is preferred and see if I can get `ggsurvplot` to work with that.

Reply:
Alright. Got a better solution for this. Took a little more digging into what `survplot` was looking for me to figure out how to address it.